app/src/main/java/com/example/plantsapp/view/PlantCreationFragment.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.plantsapp.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.example.plantsapp.databinding.FragmentPlantCreationBinding
+import timber.log.Timber
+
+class PlantCreationFragment : Fragment() {
+
+    private var _binding: FragmentPlantCreationBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentPlantCreationBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(binding) {
+            btnCreationSave.setOnClickListener {
+                collectData()
+                navigateBack()
+            }
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun collectData() {
+        with(binding) {
+            Timber.d("Plant name: ${etCreationPlantName.text}")
+            Timber.d("Plant species name: ${etCreationSpeciesName.text}")
+            Timber.d("Watering frequency in days: ${etCreationWateringFrequency.text}")
+        }
+    }
+
+    private fun navigateBack() {
+        requireActivity().supportFragmentManager.popBackStack()
+    }
```

Comment:
1. create viewmodels method and just pass here all fields. log them from viewmodel and then send event to navigate
2. Also I think it is not needed to separate almost everything into methods